main.go

```diff
@@ -40,9 +44,58 @@ func main() {
 			versioncmd.Command(version),
 			workerpools.Command(),
 		},
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:    "log-format",
+				Aliases: []string{"f"},
+				Usage:   "Log format: json, text",
+				Value:   "text",
+			},
+			&cli.StringFlag{
+				Name:    "log-level",
+				Aliases: []string{"l"},
+				Usage:   "Log level: debug, info, warn, error",
+				Value:   "info",
+			},
+		},
+		Before: func(c *cli.Context) error {
+			var logger *slog.Logger
+			switch c.String("log-format") {
```

Comment:
Nit: It's kinda nicer when flags are defined vars as we can later use those to extract values from context: https://github.com/spacelift-io/spacectl/blob/main/internal/cmd/module/create_version.go#L13

You could just define them above main here 🤷🏻